ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1027 @@
+        const WHITE_BALANCE_WHITE_FLUORESCENT_LIGHT = ctru_sys::WHITE_BALANCE_WHITE_FLUORESCENT_LIGHT;
+        const WHITE_BALANCE_DAYLIGHT                = ctru_sys::WHITE_BALANCE_DAYLIGHT;
+        const WHITE_BALANCE_CLOUDY                  = ctru_sys::WHITE_BALANCE_CLOUDY;
+        const WHITE_BALANCE_HORIZON                 = ctru_sys::WHITE_BALANCE_HORIZON;
+        const WHITE_BALANCE_SHADE                   = ctru_sys::WHITE_BALANCE_SHADE;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamPhotoMode: u32 {
+        const PHOTO_MODE_NORMAL    = ctru_sys::PHOTO_MODE_NORMAL;
+        const PHOTO_MODE_PORTRAIT  = ctru_sys::PHOTO_MODE_PORTRAIT;
+        const PHOTO_MODE_LANDSCAPE = ctru_sys::PHOTO_MODE_LANDSCAPE;
+        const PHOTO_MODE_NIGHTVIEW = ctru_sys::PHOTO_MODE_NIGHTVIEW;
+        const PHOTO_MODE_LETTER    = ctru_sys::PHOTO_MODE_LETTER;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamEffect: u32 {
+        const EFFECT_NONE     = ctru_sys::EFFECT_NONE;
+        const EFFECT_MONO     = ctru_sys::EFFECT_MONO;
+        const EFFECT_SEPIA    = ctru_sys::EFFECT_SEPIA;
+        const EFFECT_NEGATIVE = ctru_sys::EFFECT_NEGATIVE;
+        const EFFECT_NEGAFILM = ctru_sys::EFFECT_NEGAFILM;
+        const EFFECT_SEPIA01  = ctru_sys::EFFECT_SEPIA01;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamContrast: u32 {
+        const CONTRAST_PATTERN_01 = ctru_sys::CONTRAST_PATTERN_01;
+        const CONTRAST_PATTERN_02 = ctru_sys::CONTRAST_PATTERN_02;
+        const CONTRAST_PATTERN_03 = ctru_sys::CONTRAST_PATTERN_03;
+        const CONTRAST_PATTERN_04 = ctru_sys::CONTRAST_PATTERN_04;
+        const CONTRAST_PATTERN_05 = ctru_sys::CONTRAST_PATTERN_05;
+        const CONTRAST_PATTERN_06 = ctru_sys::CONTRAST_PATTERN_06;
+        const CONTRAST_PATTERN_07 = ctru_sys::CONTRAST_PATTERN_07;
+        const CONTRAST_PATTERN_08 = ctru_sys::CONTRAST_PATTERN_08;
+        const CONTRAST_PATTERN_09 = ctru_sys::CONTRAST_PATTERN_09;
+        const CONTRAST_PATTERN_10 = ctru_sys::CONTRAST_PATTERN_10;
+        const CONTRAST_PATTERN_11 = ctru_sys::CONTRAST_PATTERN_11;
+
+        const CONTRAST_LOW    = ctru_sys::CONTRAST_LOW;
+        const CONTRAST_NORMAL = ctru_sys::CONTRAST_NORMAL;
+        const CONTRAST_HIGH   = ctru_sys::CONTRAST_HIGH;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamLensCorrection: u32 {
+        const LENS_CORRECTION_OFF   = ctru_sys::LENS_CORRECTION_OFF;
+        const LENS_CORRECTION_ON_70 = ctru_sys::LENS_CORRECTION_ON_70;
+        const LENS_CORRECTION_ON_90 = ctru_sys::LENS_CORRECTION_ON_90;
+
+        const LENS_CORRECTION_DARK = ctru_sys::LENS_CORRECTION_DARK;
+        const LENS_CORRECTION_NORMAL = ctru_sys::LENS_CORRECTION_NORMAL;
+        const LENS_CORRECTION_BRIGHT = ctru_sys::LENS_CORRECTION_BRIGHT;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamOutputFormat: u32 {
```

Comment:
You should implement `TryFrom` here to handle `FramebufferFormat` to `CamOutputFormat` conversions.